include/audiothread_cmd.h

```diff
@@ -0,0 +1,605 @@
+    /* 0x14 */ AUDIOCMD_OP_CHANNEL_SET_GAIN,
+    /* 0x41 */ AUDIOCMD_OP_SEQPLAYER_FADE_VOLUME_SCALE = 0x41,
+    /* 0x46 */ AUDIOCMD_OP_SEQPLAYER_SET_IO = 0x46,
+    /* 0x47 */ AUDIOCMD_OP_SEQPLAYER_SET_TEMPO,
+    /* 0x48 */ AUDIOCMD_OP_SEQPLAYER_SET_TRANSPOSITION,
+    /* 0x49 */ AUDIOCMD_OP_SEQPLAYER_CHANGE_TEMPO,
+    /* 0x4A */ AUDIOCMD_OP_SEQPLAYER_FADE_TO_SET_VOLUME,
+    /* 0x4B */ AUDIOCMD_OP_SEQPLAYER_FADE_TO_SCALED_VOLUME,
+    /* 0x4C */ AUDIOCMD_OP_SEQPLAYER_RESET_VOLUME,
+    /* 0x4D */ AUDIOCMD_OP_SEQPLAYER_SET_BEND,
+    /* 0x4E */ AUDIOCMD_OP_SEQPLAYER_CHANGE_TEMPO_TICKS,
+    /* 0x81 */ AUDIOCMD_OP_GLOBAL_SYNC_LOAD_SEQ_PARTS = 0x81,
+    /* 0x82 */ AUDIOCMD_OP_GLOBAL_INIT_SEQPLAYER,
+    /* 0x83 */ AUDIOCMD_OP_GLOBAL_DISABLE_SEQPLAYER,
+    /* 0x85 */ AUDIOCMD_OP_GLOBAL_INIT_SEQPLAYER_SKIP_TICKS = 0x85,
+    /* 0x90 */ AUDIOCMD_OP_GLOBAL_SET_CHANNEL_MASK = 0x90,
+    /* 0xE0 */ AUDIOCMD_OP_GLOBAL_SET_DRUM_FONT = 0xE0,
+    /* 0xE1 */ AUDIOCMD_OP_GLOBAL_SET_SFX_FONT,
+    /* 0xE2 */ AUDIOCMD_OP_GLOBAL_SET_INSTRUMENT_FONT,
+    /* 0xE3 */ AUDIOCMD_OP_GLOBAL_POP_PERSISTENT_CACHE,
+    /* 0xE4 */ AUDIOCMD_OP_GLOBAL_SET_CUSTOM_FUNCTION, // TODO: check
+    /* 0xE5 */ AUDIOCMD_OP_GLOBAL_E5, // TODO: check
+    /* 0xE6 */ AUDIOCMD_OP_GLOBAL_SET_REVERB_DATA,
+    /* 0xF0 */ AUDIOCMD_OP_GLOBAL_SET_SOUND_MODE = 0xF0,
+    /* 0xF1 */ AUDIOCMD_OP_GLOBAL_MUTE,
+    /* 0xF2 */ AUDIOCMD_OP_GLOBAL_UNMUTE,
+    /* 0xF3 */ AUDIOCMD_OP_GLOBAL_SYNC_LOAD_INSTRUMENT,
+    /* 0xF4 */ AUDIOCMD_OP_GLOBAL_ASYNC_LOAD_SAMPLE_BANK,
+    /* 0xF5 */ AUDIOCMD_OP_GLOBAL_ASYNC_LOAD_FONT,
+    /* 0xF6 */ AUDIOCMD_OP_GLOBAL_DISCARD_SEQ_FONTS,
+    /* 0xF8 */ AUDIOCMD_OP_GLOBAL_STOP_AUDIOCMDS = 0xF8,
+    /* 0xF9 */ AUDIOCMD_OP_GLOBAL_RESET_AUDIO_HEAP,
+    /* 0xFA */ AUDIOCMD_OP_GLOBAL_NOOP_1, // used but no code exists for it
+    /* 0xFB */ AUDIOCMD_OP_GLOBAL_SET_CUSTOM_UPDATE_FUNCTION,
+    /* 0xFC */ AUDIOCMD_OP_GLOBAL_ASYNC_LOAD_SEQ,
+    /* 0xFD */ AUDIOCMD_OP_GLOBAL_NOOP_2, // used but no code exists for it
+    /* 0xFE */ AUDIOCMD_OP_GLOBAL_DISABLE_ALL_SEQPLAYERS
+} AudioThreadCmdOp;
+
+// ==== Audio Thread Channel Commands ====
+
+/**
+ * Set the volumeScale on a given channel
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param channelIndex the index of the channel to modify
+ * @param volumeScale (f32) the volume scale for the sequence. No change in volume is 1.0f
+ */
+#define AUDIOCMD_CHANNEL_SET_VOL_SCALE(seqPlayerIndex, channelIndex, volumeScale)                             \
+    AudioThread_QueueCmdF32(AUDIO_MK_CMD(AUDIOCMD_OP_CHANNEL_SET_VOL_SCALE, seqPlayerIndex, channelIndex, 0), \
+                            volumeScale)
+
+/**
+ *  Set the volume on a given channel
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param channelIndex the index of the channel to modify
+ * @param volume (f32) the target volume for the sequence. Default volume is 1.0f
```

Comment:
I've been contemplating this as well. Basically, every set of args is broken into 4 u8's + 1 variable sized arg:

```c
void AudioThread_QueueCmdF32(u32 opArgs, f32 data);
void AudioThread_QueueCmdS32(u32 opArgs, s32 data);
void AudioThread_QueueCmdS8(u32 opArgs, s8 data);
void AudioThread_QueueCmdU16(u32 opArgs, u16 data);
```
Where the `opArgs` is always 4 equally-sized 8-bit args, and the `data` is changeable. I've commented on the size of every `data` arg, but not every instruction even uses the `data` arg, so it's not on every set.

Not really sure what to do about it, thoughts are welcome

Reply:
So all these indices are always 8 bits?

Reply:
Yeah, it's fundamentally designed so that `opArgs` is always broken into `opArgs->op`, `opArgs->arg0`, `opArgs->arg1`, `opArgs->arg2`, so there's no variable on type. It's on a single arg that goes into data that can vary. And the audio thread command chooses one of the 4 commands I have above so it can also store 1 variable of the type it needs

Reply:
Right, so literally everything unlabelled here is a u8? I think it needs some sort of overall comment on the structure like you've just explained, but then we'd still have to decide whether to annotate every single one.